File: src/main/java/fi/finlit/edith/ui/components/InfoMessage.java
package fi.finlit.edith.ui.components;

import org.apache.tapestry5.Block;
import org.apache.tapestry5.annotations.Property;
import org.apache.tapestry5.ioc.Messages;
import org.apache.tapestry5.ioc.annotations.Inject;

public class InfoMessage {

    @Inject
    @Property
    private Block infoMessageBlock;

    @Inject
    @Property
    private Messages messages;

    @Property
    private String info;

    @Property
    private String error;

    public Block getBlock() {
        return infoMessageBlock;
    }

    public void addInfoMsg(String key) {
        info = key;
    }

    public String getInfoMsg() {
        return info != null ? messages.get(info) : "";
    }

    public String getErrorMsg() {
        return error != null ? messages.get(error) : "";
    }

    public void addErrorMsg(String key) {
        error = key;
    }



}

File: src/main/scala/fi/finlit/edith/ui/components/InfoMessage.scala
package fi.finlit.edith.ui.components

import org.apache.tapestry5.Block
import org.apache.tapestry5.annotations.Property
import org.apache.tapestry5.ioc.Messages
import org.apache.tapestry5.ioc.annotations.Inject
//remove if not needed
import scala.collection.JavaConversions._

class InfoMessage {

  @Inject
  @Property
  private var infoMessageBlock: Block = _

  @Inject
  @Property
  private var messages: Messages = _

  @Property
  private var info: String = _

  @Property
  private var error: String = _

  def getBlock(): Block = infoMessageBlock

  def addInfoMsg(key: String) {
    info = key
  }

  def getInfoMsg(): String = {
    if (info != null) messages.get(info) else ""
  }

  def getErrorMsg(): String = {
    if (error != null) messages.get(error) else ""
  }

  def addErrorMsg(key: String) {
    error = key
  }
}
